### DISCUSSION TITLE
How to use Whisper only to detect no_speech probability?

### DISCUSSION THREAD

**Discussion by rusty-ai** [openai/whisper#1935] [Q&A]
Hi, I am only using Whisper to detect the langauge using that

```
# make log-Mel spectrogram and move to the same device as the model
mel = whisper.log_mel_spectrogram(audio).to(model.device)

# detect the spoken language
_, probs = model.detect_language(mel)
print(f"Detected language: {max(probs, key=probs.get)}")
```

but it only returns probabilities, that the given 30 sec chunk of audio has the specific language, but it does not return the no_speech probability. Is there any way to only return the no_speech probability, without using .transcribe, which is computationally expensive?

**Comment by glangford** [comment]
It would be better to use a dedicated voice activity detector, such as Silero VAD. The no_speech probability in whisper is not very accurate...but if you want to use just whisper, see this discussion

* https://github.com/openai/whisper/discussions/96
